internal/enginecache/cache.go

```diff
@@ -360,24 +376,80 @@ func ValidateOrUpdateEngineAtPath(targetOS string, cachePath string) (engineCach
 		createSymLink("Versions/Current/Headers", frameworkDestPath+"/Headers")
 		createSymLink("Versions/Current/Modules", frameworkDestPath+"/Modules")
 		createSymLink("Versions/Current/Resources", frameworkDestPath+"/Resources")
+	} else {
+		// Any other engine than darwin JIT
+		for _, engineFile := range build.EngineFiles(targetOS, mode) {
+			err := moveFile(
+				filepath.Join(engineExtractPath, engineFile),
+				filepath.Join(engineCachePath, engineFile),
+			)
+			if err != nil {
+				log.Errorf("Failed to move downloaded %s: %v", engineFile, err)
+				os.Exit(1)
+			}
+		}
+	}
 
-	case "linux-x64":
-		err := moveFile(
-			filepath.Join(engineExtractPath, "libflutter_engine.so"),
-			filepath.Join(engineCachePath, "/libflutter_engine.so"),
-		)
+	// Strip linux JIT engine after download and not at every build
+	if targetOS == "linux" && !mode.IsAot {
+		err = exec.Command("strip", "-s", filepath.Join(engineCachePath, build.EngineFiles(targetOS, mode)[0])).Run()
 		if err != nil {
-			log.Errorf("Failed to move downloaded libflutter_engine.so: %v", err)
+			log.Errorf("Failed to strip %s: %v", filepath.Join(engineCachePath, build.EngineFiles(targetOS, mode)[0]), err)
```

Comment:
`filepath.Join(engineCachePath, build.EngineFiles(targetOS, mode)[0])` can be placed in a local var